__tests__/happy.dlstreamConnection.js

```diff
@@ -0,0 +1,24 @@
+import 'dotenv/config';
+
+import onErrorResumeNext from 'on-error-resume-next';
+
+import { timeouts } from './constants.json';
+import * as createDirectLine from './setup/createDirectLine';
+import waitForConnected from './setup/waitForConnected';
+
+describe('test dl streaming end', () => {
+  let unsubscribes;
+  let directLine;
+  const ConnectionStatusEnd = 5;
+  beforeEach(() => unsubscribes = []);
+  afterEach(() => unsubscribes.forEach(fn => onErrorResumeNext(fn)));
+  test('using Streaming Extensions', async () => {
+    jest.setTimeout(30000);
```

Comment:
I think we need to expand this test a bit to make sure "when connection status ended, it should not retry."

I am not blocking the PR. Please add the following comment:

```js
// TODO: [hawo] Need more realistic testing.
//       - Able to connect to a Web Socket server
//       - Make sure after `end` is called, the client will not reconnect
//       - If the connection is disrupted, make sure the client will reconnect
```